keyboards/chromatonemini/keymaps/led/keymap.c

```diff
@@ -0,0 +1,687 @@
+            MI_BENDD,           MI_B_4, MI_A_4, MI_G_4,  MI_F_4,  MI_Eb_4,  MI_Db_4, MI_B_3, MI_A_3, MI_G_3,  MI_F_3, MI_Eb_3,  MI_Db_3, MI_B_2, MI_A_2, MI_G_2,  MI_F_2,  MI_Eb_2,  MI_Db_2      \
+    ),
+
+    /* Flip Trans   This layer must locate 1 layer above _FLIPBASE layer.  MI_TRNSU and MI_TRNSD are swapped. */
+    [_FLIPTRANS] = LAYOUT_with_sustain_pedal( \
+            _______,          _______,                                                                                                                                                                   \
+            _______,                                                                                                                                                                                     \
+        MI_TRNSU, MI_TRNSD,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+            _______,               _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______      \
+    ),
+
+    [_FN] =  LAYOUT_with_sustain_pedal( \
+            _______,          _______,                                                                                                                                                                   \
+            MI_VELU,                                                                                                                                                                                     \
+        MI_OCTD, MI_OCTU,     B_BASE,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, B_LEFT, XXXXXXX, XXXXXXX, B_CENTER, XXXXXXX, XXXXXXX, B_RIGHT, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, B_FLIP,  \
+            MI_VELD,               TGLINTR, TGLTRNS, TGLCHGR, XXXXXXX, XXXXXXX, RGB_SAD, RGB_SAI, RGB_HUD, RGB_HUI, RGB_SPD, RGB_SPI, RGB_VAD, RGB_VAI, RGB_RMOD, RGB_MOD, EEP_RST, TGLINDI, RGB_TOG     \
+    )
+};
+
+// #ifndef CONSOLE_ENABLE
+void eeconfig_init_user(void) {  // EEPROM is getting reset!
+    rgb_matrix_enable();
+    rgb_matrix_set_speed(RGB_MATRIX_STARTUP_SPD);
+    rgb_matrix_sethsv(HSV_DARKBLUE);
+    // rgb_matrix_mode(RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE);
+    // rgb_matrix_mode(RGB_MATRIX_CUSTOM_my_solid_reactive_multiwide_col);
+    rgb_matrix_mode(RGB_MATRIX_CUSTOM_my_solid_reactive_col);
+}
+// #endif
+
+void keyboard_post_init_user(void) {
+    //  Set octave to MI_OCT_1
+    midi_config.octave = MI_OCT_0 - MIDI_OCTAVE_MIN;
+
+    // avoid using 127 since it is used as a special number in some sound sources.
+    midi_config.velocity = MIDI_INITIAL_VELOCITY;
+
+    for (uint8_t i = 0; i < MY_TONE_COUNT; i++) {
+        my_tone_status[i] = MIDI_INVALID_NOTE;
+    }
+
+    default_layer_set(_LS_BASE);
+    layer_state_set(_LS_BASE);
```

Comment:
Neither of these should be needed, as it should happen on init.

Reply:
This is mandatory since both default_layer_state and layer_state are initialized to zero here.
https://github.com/qmk/qmk_firmware/blob/495d47b7d2a0df4e84c25c9cfee3931576cfbd32/tmk_core/common/action_layer.c#L15
https://github.com/qmk/qmk_firmware/blob/495d47b7d2a0df4e84c25c9cfee3931576cfbd32/tmk_core/common/action_layer.c#L81

Here, it will change them to  
(1UL << _BASE) 
= (1UL << 0) 
= 1.

It indeed works unexpectedly if I comment them out.

Reply:
But both layer_state_set(0) and default_layer_set(0) are called during init. 
https://github.com/qmk/qmk_firmware/blob/cca76beb2e996ef878c72765adb1245f82f04329/quantum/bootmagic/magic.c#L50-L54

Reply:
Sorry for the confusion, what is expected to do here is to set layer_state_set(_1_) and default_layer_set(_1_).
That’s why these rows are needed.